lib/ui/popup.js

```diff
@@ -20,12 +21,27 @@ function Popup(opts) {
 
   this.beforePopupHide = this.beforePopupHide.bind(this);
   this.onResultsMouseOver = this.onResultsMouseOver.bind(this);
+  this.onFirstPopupOpen = this.onFirstPopupOpen.bind(this);
 }
 
 Popup.prototype = {
   init: function() {
     this.el = this.win.document.getElementById('PopupAutoCompleteRichResult');
     this.el.addEventListener('popuphiding', this.beforePopupHide);
+
+    // The first time the popup opens, a bit of popup state is not set
+    // correctly, causing no results to be shown, and causing the popup to fail
+    // to appear more than once (#138). To work around this bug, listen for the
+    // popup to be opened for the first time, and set the missing state.
+    //
```

Comment:
Perhaps a link to the relevant Gecko code this is emulating would be nice for future maintainers?